src/devices/Vl53L1X/Vl53L1X.cs

```diff
@@ -0,0 +1,1022 @@
+            ushort spNb = ReadUInt16((ushort)Registers.VL53L1X_RESULT__DSS_ACTUAL_EFFECTIVE_SPADS_SD0);
+            return (ushort)(2000.0 * signal / spNb);
+        }
+
+        /// <summary>
+        ///     This function returns the ambient per SPAD in kcps/SPAD.
+        /// </summary>
+        /// <returns>The ambient per SPAD in kcps/SPAD.</returns>
+        public ushort GetAmbientPerSpad()
+        {
+            ushort ambientRate = ReadUInt16((ushort)Registers.RESULT__AMBIENT_COUNT_RATE_MCPS_SD);
+            ushort spNb = ReadUInt16((ushort)Registers.VL53L1X_RESULT__DSS_ACTUAL_EFFECTIVE_SPADS_SD0);
+            return (ushort)(2000.0 * ambientRate / spNb);
+        }
+
+        /// <summary>
+        ///     This function returns the signal in kcps.
+        /// </summary>
+        /// <returns>the signal in kcps.</returns>
+        public ushort GetSignalRate()
+        {
+            return (ushort)(ReadUInt16((ushort)Registers
+                .VL53L1X_RESULT__PEAK_SIGNAL_COUNT_RATE_CROSSTALK_CORRECTED_MCPS_SD0) * 8);
+        }
+
+        /// <summary>
+        ///     This function returns the current number of enabled SPADs.
+        /// </summary>
+        /// <returns>the current number of enabled SPADs.</returns>
+        public ushort GetSpadNb()
+        {
+            return (ushort)(ReadUInt16((ushort)Registers.VL53L1X_RESULT__DSS_ACTUAL_EFFECTIVE_SPADS_SD0) >> 8);
+        }
+
+        /// <summary>
+        ///     This function returns the ambient rate in kcps.
+        /// </summary>
+        /// <returns>The ambient rate in kcps.</returns>
+        public ushort GetAmbientRate()
+        {
+            return (ushort)(ReadUInt16((ushort)Registers.RESULT__AMBIENT_COUNT_RATE_MCPS_SD) * 8);
+        }
+
+        /// <summary>
+        ///     This function returns the <see cref="RangeStatus" /> of the device.
+        /// </summary>
+        /// <returns>The <see cref="RangeStatus" /></returns>
+        public RangeStatus GetRangeStatus()
+        {
+            byte rgSt = ReadByte((ushort)Registers.VL53L1X_RESULT__RANGE_STATUS);
+
+            rgSt = (byte)(rgSt & 0x1F);
+
+            switch (rgSt)
+            {
+                case 9:
+                    return RangeStatus.NoError;
+                case 6:
+                    return RangeStatus.SigmaFailure;
+                case 4:
+                    return RangeStatus.SignalFailure;
+                case 5:
+                    return RangeStatus.OutOfBounds;
+                case 7:
+                    return RangeStatus.WrapAround;
+                default:
+                    throw new Exception($"The returned range status code {rgSt} of the device is unknown.");
+            }
+        }
+
+        /// <summary>
+        ///     This function programs the offset correction in mm.
+        /// </summary>
+        /// <param name="offsetValue">The offset correction value to program in mm</param>
+        public void SetOffset(short offsetValue)
```

Comment:
this one is a great candidate for a property as well

Reply:
I'm now using properties wherever possible